gpii/node_modules/flowManager/src/FlowManager.js

```diff
@@ -136,9 +133,13 @@ fluid.defaults("gpii.flowManager.local", {
             record: "gpii.lifecycleManager.userLogonHandling.matchMakingStateChangeHandler",
             target: "{that gpii.lifecycleManager.loginRequest}.options.gradeNames"
         },
-        "userLogonHandling.defaultSettingsDataPromise": {
-            record: "{gpii.flowManager.local}.defaultSettingsDataPromise",
-            target: "{that gpii.lifecycleManager.userLogonHandling.stateChangeHandler}.options.members.defaultSettingsDataPromise"
+        "userLogonHandling.defaultSnapshot": {
+            record: "{gpii.flowManager.local}.defaultSnapshot",
```

Comment:
This is unconvincing - the member is not initialised until the execution of convertDefaultSettingsToSnapshot and so the risk of a race remains. I think it would be safest to remove all of these volatile members and just distribute references to the promises instead.